File: Chapter09/DisplayPlatformInfo/DisplayPlatformInfo/DisplayPlatformInfo.iOS/PlatformInfo.cs
using System;
using UIKit;
using Xamarin.Forms;

[assembly: Dependency(typeof(DisplayPlatformInfo.iOS.PlatformInfo))]

namespace DisplayPlatformInfo.iOS
{
    public class PlatformInfo : IPlatformInfo
    {
        UIDevice device = new UIDevice();

        public string GetModel()
        {
            return device.Model.ToString();
        }

        public string GetVersion()
        {
            return String.Format("{0} {1}", device.SystemName, 
                                            device.SystemVersion);
        }
    }
}

File: Chapter09/FS/DisplayPlatformInfo/iOS/PlatformInfo.fs
namespace DisplayPlatformInfo.iOS

open UIKit
open System
open DisplayPlatformInfo
open Foundation
open Xamarin.Forms
open Xamarin.Forms.Xaml

type PlatformInfo () =
    let device = new UIDevice()

    interface IPlatformInfo with
        member this.GetModel () = device.Model.ToString()

        member this.GetVersion () = sprintf "%s %s" device.SystemName device.SystemVersion

module DisplayPlatform =
    [<assembly: Dependency(typeof<PlatformInfo>)>]
    do()




